cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CleanCommand.java

```diff
@@ -93,18 +93,41 @@ public void execute() {
             CommandUtil.printError(this.outStream,
                     "provided target directory '" + this.targetDir + "' does not exist.",
                     null, false);
+        } else if (!Files.isDirectory(this.targetDir)) {
+            CommandUtil.printError(this.outStream,
+                    "provided target path '" + this.targetDir + "' is not a directory.",
+                    null, false);
+        } else {
+            ProjectUtils.deleteDirectory(this.targetDir);
+            this.outStream.println("Successfully deleted '" + this.targetDir + "'.");
+        }
+
+        //delete the generated directory
+        Path generatedDir;
+        try {
+            Project project = BuildProject.load(this.projectPath);
+            generatedDir = project.sourceRoot().resolve(ProjectConstants.GENERATED_MODULES_ROOT);
+        } catch (ProjectException e) {
+            CommandUtil.printError(this.outStream, e.getMessage(), null, false);
             CommandUtil.exitError(this.exitWhenFinish);
             return;
         }
-        if (!Files.isDirectory(this.targetDir)) {
+        if (Files.notExists(generatedDir)) {
             CommandUtil.printError(this.outStream,
-                    "provided target path '" + this.targetDir + "' is not a directory.",
+                    "generated directory '" + generatedDir + "' does not exist.",
```

Comment:
We can just ignore it if the generated/ directory does not exist. We do not need to fail.